server/app/mutations/grid_services/common.rb

```diff
@@ -172,7 +171,7 @@ def common_validations
           end
           string :entrypoint
           array :env do
-            string
+            string matches: /\A\S+(?<!\\)=.*/
```

Comment:
It's from #2241, and matches what's used for `GridService#env_hash`. Perhaps it should be some `GridService.ENV_REGEXP` constant?